applications/ChimeraApplication/custom_processes/apply_chimera_process.cpp

```diff
@@ -0,0 +1,627 @@
+        r_hole_model_part.CreateSubModelPart(mBoundaryName);
+
+    auto& r_patch_boundary_model_part = ExtractPatchBoundary(
+        PatchParameters, r_background_boundary_model_part, DomainType);
+
+    BuiltinTimer bg_distance_calc_time;
+    DistanceCalculationUtility<TDim>::CalculateDistance(
+        r_background_model_part, r_patch_boundary_model_part);
+    KRATOS_INFO_IF(
+        "Distance calculation on background took                  : ", mEchoLevel > 0)
+        << bg_distance_calc_time.ElapsedSeconds() << " seconds" << std::endl;
+
+    BuiltinTimer hole_creation_time;
+    ChimeraHoleCuttingUtility().CreateHoleAfterDistance<TDim>(
+        r_background_model_part, r_hole_model_part, r_hole_boundary_model_part, over_lap_distance);
+    KRATOS_INFO_IF(
+        "ApplyChimera : Hole creation took                        : ", mEchoLevel > 0)
+        << hole_creation_time.ElapsedSeconds() << " seconds" << std::endl;
+
+    const int n_elements = static_cast<int>(r_hole_model_part.NumberOfElements());
+#pragma omp parallel for
+    for (int i_elem = 0; i_elem < n_elements; ++i_elem) {
+        ModelPart::ElementsContainerType::iterator it_elem =
+            r_hole_model_part.ElementsBegin() + i_elem;
+        it_elem->Set(VISITED, true); // for multipatch
+    }
+
+    BuiltinTimer mpc_time;
+    ApplyContinuityWithMpcs(r_patch_boundary_model_part, *p_point_locator_on_background);
+    ApplyContinuityWithMpcs(r_hole_boundary_model_part, *p_pointer_locator_on_patch);
+    KRATOS_INFO_IF(
+        "ApplyChimera : Creation of MPC for chimera took          : ", mEchoLevel > 0)
+        << mpc_time.ElapsedSeconds() << " seconds" << std::endl;
+
+    r_hole_boundary_model_part.RemoveSubModelPart(mBoundaryName);
+    r_background_model_part.RemoveSubModelPart(mHoleName);
+    r_patch_model_part.RemoveSubModelPart(mModifiedName);
+}
+
+template <int TDim>
+void ApplyChimera<TDim>::CreateConstraintIds(std::vector<int>& rIdVector,
+                                             const ApplyChimera<TDim>::IndexType NumberOfConstraintsRequired)
+{
+    ApplyChimera<TDim>::IndexType max_constraint_id = 0;
+    // Get current maximum constraint ID
+    if (mrMainModelPart.MasterSlaveConstraints().size() != 0) {
+        mrMainModelPart.MasterSlaveConstraints().Sort();
+        ModelPart::MasterSlaveConstraintContainerType::iterator it =
+            mrMainModelPart.MasterSlaveConstraintsEnd() - 1;
+        max_constraint_id = (*it).Id();
+        ++max_constraint_id;
+    }
+
+    // Now create a vector size NumberOfConstraintsRequired
+    rIdVector.resize(NumberOfConstraintsRequired * (TDim + 1));
+    std::iota(std::begin(rIdVector), std::end(rIdVector), max_constraint_id); // Fill with consecutive integers
+}
+
+template <int TDim>
+void ApplyChimera<TDim>::ApplyContinuityWithMpcs(ModelPart& rBoundaryModelPart,
+                                                 ApplyChimera<TDim>::PointLocatorType& pBinLocator)
+{
+}
```

Comment:
Are you planning to implement this in the future?

Reply:
This is the function that is overwritten by the Monolith And FractionalStep processes.  If it confuses you I can remove the empty implementation here !